src/opening-period/rule/Rule.tsx

```diff
@@ -0,0 +1,303 @@
+                modifierOption.value === currentFrequency.frequency_modifier
+            )?.label
+          ),
+          value: currentFrequency,
+        },
+      ];
+
+  const frequencyOptions: InputOption[] = allFrequencyValues.map(
+    convertFrequencyToInputOption
+  );
+
+  const frequencyModifierField = `rules[${index}].frequency_modifier`;
+  const frequencyOrdinalField = `rules[${index}].frequency_ordinal`;
+
+  const onFrequencyChange = (selected: InputOption): void => {
+    const selectedFrequency = allFrequencyValues.find(
+      ({ label }) => label === selected.label
+    );
+
+    setValue(
+      frequencyModifierField,
+      selectedFrequency?.value.frequency_modifier || null
+    );
+    setValue(
+      frequencyOrdinalField,
+      selectedFrequency?.value.frequency_ordinal || null
+    );
+  };
+
+  useEffect(() => {
+    register({ name: frequencyModifierField });
+    register({ name: frequencyOrdinalField });
+
+    // "If you choose to not use Controller and manually register fields, you will need to update the input value with setValue." - https://react-hook-form.com/api#register
+    setValue(frequencyModifierField, frequencyModifier, {
+      shouldValidate: false,
+    });
+    setValue(frequencyOrdinalField, frequencyOrdinal, {
+      shouldValidate: false,
+    });
+  }, [
+    frequencyModifier,
+    frequencyModifierField,
+    frequencyOrdinal,
+    frequencyOrdinalField,
+    register,
+    setValue,
+  ]);
+
+  return (
+    <div className="opening-group-rule form-control" key={`rules-${index}`}>
+      <input
+        type="hidden"
+        name={`rules[${index}].id`}
+        defaultValue={id}
+        ref={register()}
+      />
+      <div className="opening-group-rule-remove">
+        <HDSButton
+          data-test={`remove-rule-button-${index}`}
+          className="opening-period-remove-list-item-button"
+          variant="supplementary"
+          onClick={(): void => remove(index)}
+          iconLeft={<IconTrash />}>
+          Poista aukioloaikojen voimassaolosääntö
+        </HDSButton>
+      </div>
+      <div className="opening-group-rule-fieldset">
+        <Controller
+          key={`rule-${index}-context`}
+          name={`rules[${index}].context`}
+          control={control}
+          defaultValue={`${context || ''}`}
+          render={({ onChange, value }): JSX.Element => (
+            <Select
+              id={`rule-${index}-context`}
+              className="opening-group-rule-column opening-group-rule-select"
+              onChange={(selected: InputOption): void =>
+                onChange(selected.value)
+              }
+              options={contextOptions}
+              defaultValue={contextOptions.find(
+                (option: InputOption): boolean => option.value === value
+              )}
+              label="Valitse aika"
+              placeholder="Aikaväli"
+            />
+          )}
+        />
+        <Select
```

Comment:
Why is this select not wrapped inside Controller, but the other Select components are?

Reply:
It is an input that is derivated from two values. It requires that we need to register two values, and it is possible with control. Maybe I could extract it into a component because of its different behavior.

Reply:
Thanks for clarification. Probably no need to extract it imo, it's better inlined